### PR TITLE
Add application date to Person

### LINKED ISSUE #64: Implement ApplicationDateTime class
So that HMHero can track the date of application for applicants for usage in other commands

### PR DESCRIPTION
**Pull request by Jxleejiaxin**
Refer to #66 
## Enhancement:
If `/applied` prefix is used when adding applicant, user can specify application time. If prefix is not used, HMHero will set current time as application time.

Fix #64

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Person.java [hunk 1]
```diff
@@ -34,14 +32,15 @@ public class Person {
...
+                  ApplicationDateTime applicationDateTime, Optional<InterviewDateTime> interviewDateTime,
+                  Set<Note> notes) {
         requireAllNonNull(name, phone, email, address, notes);
         this.name = name;
         this.phone = phone;
         this.email = email;
         this.address = address;
         this.status = status;
-        this.applicationDateTime = new ApplicationDateTime(LocalDateTime.now().truncatedTo(ChronoUnit.MINUTES));
+        this.applicationDateTime = applicationDateTime;
```
Comment (Merrickneo) [inline src/main/java/seedu/address/model/person/Person.java:43]: Good way to help users edit the application date according to their needs

**Review by Merrickneo — APPROVED** [review]
LGTM! Good use of the `applied/` tag to help users manage the application date of the `Applicant`.

File: src/main/java/seedu/address/model/person/Person.java (hunk 1 above)
Comment (FredericChow00) [reply src/main/java/seedu/address/model/person/Person.java:43]: Agreed

**Review by FredericChow00 — APPROVED** [review]
LGTM!

**Pull request merged**